src/main/java/hk/com/novare/hellogit/services/UserFilesService.java

```diff
@@ -80,6 +80,18 @@ public User deleteUserByName(String name) {
         return null;
     }
 
+    public User updateUserByName(String name, User newUser){
+        for (User u : users) {
+            if (u.getName().equals(name)) {
+                newUser.setName(name);
```

Comment:
This will only "update" the name. What if I wanted to update other properties?

You may want to use [`BeanUtils#copyProperties`](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/beans/BeanUtils.html#copyProperties-java.lang.Object-java.lang.Object-) to properly duplicate the existing user, then apply the updated changes from the  user's request to the copied original.